base_location_nuts/models/res_partner.py

```diff
@@ -1,61 +1,123 @@
+
+    @api.multi
+    def _onchange_nuts(self, level):
+        field = self["nuts%d_id" % level]
+        country_id = field.country_id.id
+        state_id = field.state_id.id
+        if country_id and self.country_id.id != country_id:
+            self.country_id = country_id
+        if state_id and self.state_id.id != state_id:
+            self.state_id = state_id
+        if (level - 1) > 0:
+            parent_id = field.parent_id.id
+            if parent_id:
+                parent_field = 'nuts%d_id' % (level - 1)
+                if self[parent_field].id != parent_id:
+                    self[parent_field] = parent_id
+        result = dict()
+        if country_id and level < 4:
+            result['domain'] = dict()
+            while level < 4:
+                parent_field = 'nuts%d_id' % level
+                domain_field = 'nuts%d_id' % (level + 1)
+                parent_id = self[parent_field].id
+                if parent_id:
+                    result['domain'][domain_field] = [
+                        ('parent_id', '=', parent_id),
+                    ]
+                level += 1
+        return result
+
+    @api.multi
+    @api.onchange('nuts4_id')
+    def _onchange_nuts4_id(self):
+        return self._onchange_nuts(4)
+
+    @api.multi
+    @api.onchange('nuts3_id')
+    def _onchange_nuts3_id(self):
+        return self._onchange_nuts(3)
+
+    @api.multi
+    @api.onchange('nuts2_id')
+    def _onchange_nuts2_id(self):
+        return self._onchange_nuts(2)
+
+    @api.multi
+    @api.onchange('nuts1_id')
+    def _onchange_nuts1_id(self):
+        return self._onchange_nuts(1)
 
-    @api.one
-    @api.depends('country_id')
-    def _labels_get(self):
-        self.lbl_region = _('Region')
-        self.lbl_substate = _('Substate')
+    @api.multi
+    @api.onchange('country_id')
+    def _onchange_country_id(self):
+        """Sensible values and domains for related fields."""
+        fields = ['state_id', 'nuts1_id', 'nuts2_id', 'nuts3_id', 'nuts4_id']
+        country_domain = ([('country_id', '=', self.country_id.id)]
+                          if self.country_id else [])
+        domain = dict()
+        for field in fields:
+            if self.country_id and self[field].country_id != self.country_id:
+                self[field] = False
+            domain[field] = list(country_domain)  # Using list() to copy
+        fields.remove('state_id')
+        for field in fields:
+            level = int(field[:5][-1])
+            if level:
+                domain[field].append(('level', '=', level))
+        if self.country_id:
+            nuts1 = self.env['res.partner.nuts'].search([
+                ('level', '=', 1),
+                ('country_id', '=', self.country_id.id),
+            ], limit=1)
+            if self.nuts1_id.id != nuts1.id:
+                self.nuts1_id = nuts1.id
+        return {
+            'domain': domain,
+        }
 
     @api.multi
     def onchange_state(self, state_id):
```

Comment:
This method is not used anymore. Please find the equivalent one or declare a new unique onchange (`def onchange_state_id_base_location_nuts(self):`.